packages/airnode-node/src/api/processing.ts

```diff
@@ -35,7 +46,17 @@ export const preProcessApiSpecifications = async (payload: ApiCallPayload): Prom
   }
 
   // Let this throw if the processed parameters are invalid
-  const parameters = apiCallParametersSchema.parse(goProcessedParameters.data);
+  let parameters = apiCallParametersSchema.parse(goProcessedParameters.data);
+
+  if (payload.type === 'http-gateway') {
+    // Add back reserved parameters for the HTTP gateway special case
+    parameters = Object.entries(aggregatedApiCall.parameters).reduce((params, [key, value]) => {
```

Comment:
This line being commented on is where reserved parameters are added back rather than removed, but I know what you mean so I added two functions: one to remove reserved parameters and one to re-insert them after processing